src/main/java/br/com/harbitech/school/category/CategoryForm.java

```diff
@@ -0,0 +1,148 @@
+    @Pattern(regexp = "[-a-z]+", message = "{category.codeUrl.pattern}")
+    private String codeUrl;
+    private String description;
+    private String studyGuide;
+    private CategoryStatus status = CategoryStatus.INACTIVE;
+    @Min(-1)
+    private int orderVisualization;
+    @Size(max = 400, message = "{category.iconPath.size.max}")
+    private String iconPath;
+    @Size(max = 7, message = "{category.htmlHexColorCode.size.max}")
+    private String htmlHexColorCode;
+
+    public CategoryForm() {}
+
+    public CategoryForm(String name, String codeUrl, CategoryStatus status, int orderVisualization,
+                        String studyGuide, String iconPath, String htmlHexColorCode, String description) {
+        this.name = name;
+        this.codeUrl = codeUrl;
+        this.description = description;
+        this.studyGuide = studyGuide;
+        this.status = status;
+        this.orderVisualization = orderVisualization;
+        this.iconPath = iconPath;
+        this.htmlHexColorCode = htmlHexColorCode;
+    }
+
+    public CategoryForm(Long id, String name, String codeUrl, CategoryStatus status, int orderVisualization,
+                        String studyGuide, String iconPath, String htmlHexColorCode, String description) {
+        this(name, codeUrl, status, orderVisualization, studyGuide, iconPath, htmlHexColorCode, description);
+        this.id = id;
+    }
+    public Long getId() {
+        return id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getCodeUrl() {
+        return codeUrl;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public String getStudyGuide() {
+        return studyGuide;
+    }
+
+    public CategoryStatus getStatus() {
+        return status;
+    }
+
+    public int getOrderVisualization() {
+        return orderVisualization;
+    }
+
+    public String getIconPath() {
+        return iconPath;
+    }
+
+    public String getHtmlHexColorCode() {
+        return htmlHexColorCode;
+    }
+
+    public void setId(Long id) {
+        this.id = id;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public void setCodeUrl(String codeUrl) {
+        this.codeUrl = codeUrl;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    public void setStudyGuide(String studyGuide) {
+        this.studyGuide = studyGuide;
+    }
+
+    public void setStatus(CategoryStatus status) {
+        this.status = status;
+    }
+
+    public void setOrderVisualization(int orderVisualization) {
+        this.orderVisualization = orderVisualization;
+    }
+
+    public void setIconPath(String iconPath) {
+        this.iconPath = iconPath;
+    }
+
+    public void setHtmlHexColorCode(String htmlHexColorCode) {
+        this.htmlHexColorCode = htmlHexColorCode;
+    }
+
+    public static Category toModel(CategoryForm categoryform) {
```

Comment:
Acho que a única mudança aqui que eu sugeriria seria tirar o `static` desse método, aí no seu controller você pode chamar por `categoryForm.toModel()` e aqui dentro desse método você usa o `this`.

Reply:
Verdade, hein, @henriqueln7!

Acho importante fazer isso, @mariasantosdev

Reply:
Muito obrigada pessoal, melhoria realizada! eu commitei lá na semana10, muito bem observado esse ponto @henriqueln7